src/ixdat/readers/cinfdata_db.py

```diff
@@ -0,0 +1,364 @@
+                obj_as_dict = self.create_spectrum(key)  # group_data, group_meta, key)
+                obj_as_dict["name"] = self.group_meta[key]["name"]
+                spectrum_list.append(self.measurement_class.from_dict(obj_as_dict))
+
+            elif self.group_meta[key]["type"] == 4:
+                self.x_name = "Mass [AMU]"
+                self.x_unit_name = "m/z"
+                self.field_name = "Current"
+                self.field_unit = "[A]"
+                self.technique = "MS_spectrum"
+                obj_as_dict = self.create_spectrum(key)
+                obj_as_dict["name"] = self.sample_name
+                spectrum_list.append(self.measurement_class.from_dict(obj_as_dict))
+
+            else:
+                pass
+
+        if not spectrum_list:
+            warnings.warn(
+                f"No spectrum was found using '{self.token}' "
+                f" and group_column: '{self.grouping_column}'",
+                stacklevel=2,
+            )
+            return None
+        elif self.mass_scans:
+            return spectrum_list
+        elif len(spectrum_list) == 1:
+            return spectrum_list[0]
+        else:
+            try:
+                return SpectrumSeries.from_spectrum_list(spectrum_list)
+            except ValueError:
+                warnings.warn(
+                    "Could not return SpectrumSeries from list of spectrums "
+                    f"using '{self.token}' and group column: "
+                    f"'{self.grouping_column}'. \n"
+                    " Return list of all Spectrums.",
+                    stacklevel=2,
+                )
+
+                return spectrum_list
+
+    def create_spectrum(self, key):
+        x_col = self.group_data[key][:, 0]
+        y_col = self.group_data[key][:, 1]
+        tstamp = self.group_meta[key]["unixtime"]
+
+        xseries = DataSeries(data=x_col, name=self.x_name, unit_name=self.x_unit_name)
+        field = Field(
+            data=y_col,
+            name=self.field_name,
+            unit_name=self.field_unit,
+            axes_series=[
+                xseries,
+            ],
+        )
+
+        obj_as_dict = {
+            "sample_name": self.sample_name,
+            "technique": self.technique,
+            "field": field,
+            "tstamp": tstamp,
+        }
+
+        return obj_as_dict
+
+    def add_mass_scans(self):
+        """Get corrosponding mass scans to mass_time from 'comment'"""
+        self.measurement_class = MSSpectrum
+        self.grouping_column = "comment"
+        self.token = self.sample_name
+        spectrum_list = self.read_spectrums()
+
+        if self.verbose:
+            print(
+                "Using ", self.measurement.time_series[-1], " to find end of experiment"
+            )
+            print(
+                "Unixtime end of exp ",
+                self.measurement.time_series[-1].data[-1] + self.tstamp,
+            )
+
+        index = -1
```

Comment:
yes. The comment has become very long and it might need to be put somewhere else?

The idea is simply to compare tstamps of all downloaded spectra with the measurment tstamp + time to find all spectras that was taken within the specific measurement.

Reply:
I think a long comment here is fine. Thanks!